### DISCUSSION TITLE
Welcome to Topiary Discussions!

### DISCUSSION THREAD

**Discussion by torhovland** [topiary/topiary#369] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by haikyuu** [comment]
Hey, I'm Abdellah. I'm a maintainer and contributor in [imba](https://imba.io) and I'm leading the effort to modernize the tooling (Vite support and integration, WIP tree-sitter grammar, [ts2imba (typescript to imba converter)](ts2imba.com))

And since we don't have a formatter yet, I want to build one using topiary. I didn't look at that in details yet since the tree sitter grammar isn't yet complete. But having [support for tabs](https://github.com/tweag/topiary/issues/105) is a little blocker since Imba is indentation (tabs) sensitive.

In the meantime, I'm very excited about this project as it aims to unify efforts across languages and tools (just like tree-sitter itself or other tools like Vite or [Volar.js](https://volarjs.github.io/))

**Reply by ErinvanderVeen** [reply]
Hej Abdellah! Thanks for your interest in Topiary!

We'll discuss #105 in today's meeting and we'll try to get on it as soon as possible.

**Comment by Gregoor** [comment]
I have dreamed of this kind of project existing and here it is, existing and prospering. I'm beyond excited! Thank you for working on this.

My intention is to use this in my structured code editing experiments. I built and am using a VSCode extension on top of babel and prettier. Unfortunately the parsing is quite slow with Babel and I have been playing with using tree-sitter for a while, but the missing piece was an AST compatible formatter. So I'm hoping to get started on a Rust rewrite using this soon.
